components/script/dom/xrinputsourceschangeevent.rs

```diff
@@ -88,12 +88,12 @@ impl XRInputSourcesChangeEvent {
         let _ac = enter_realm(global);
         let cx = GlobalScope::get_cx();
         unsafe {
-            rooted!(in(*cx) let mut added_val = UndefinedValue());
-            added.to_jsval(*cx, added_val.handle_mut());
-            changeevent.added.set(added_val.get());
-            rooted!(in(*cx) let mut removed_val = UndefinedValue());
-            removed.to_jsval(*cx, removed_val.handle_mut());
-            changeevent.removed.set(removed_val.get());
+            changeevent
+                .added
+                .set(to_frozen_array(added, JSContext::from_ptr(*cx)));
```

Comment:
Oh, we can just pass `cx` here instead of dereferencing it and recreating the JSContext type from the pointer.

Reply:
done!
also shall i remove this?
```
warning: unnecessary `unsafe` block
  --> components\script\dom\xrinputsourceschangeevent.rs:90:9
   |
90 |         unsafe {
   |         ^^^^^^ unnecessary `unsafe` block
   |
   = note: `#[warn(unused_unsafe)]` on by default
```

Reply:
Yep!